400-claims.md

```diff
@@ -104,7 +106,7 @@ The fields above are defined as follows:
 - `custom` (OPTIONAL): A section for custom extension data applicable to a given runtime.
 - `installation` (REQUIRED): The name of the _installation_. This can be automatically generated, though humans may need to interact with it. It MUST be unique within the installation environment, though that constraint MUST be imposed externally. Elsewhere, this field is referenced as the _installation name_. The format of this field must follow the same format used for the `installation` field in the [bundle.json file specification](101-bundle-json.md#the-bundlejson-file).
 - `parameters` (OPTIONAL): Key/value pairs that were passed in during the operation. These are stored so that the operation can be re-run. Some implementations MAY choose not to store these for security or portability reasons. However, there are some caveats. See the [Parameters](#parameters) section below for details.
-- `revision` (REQUIRED): An [ULID](https://github.com/ulid/spec) that MUST change each time the claim is modified. It MUST NOT change when a [non-modifying operation](https://github.com/cnabio/cnab-spec/blob/master/101-bundle-json.md#custom-actions) is performed on an installation.
+- `revision` (REQUIRED): An [ULID](https://github.com/ulid/spec) that MUST change each time the claim is modified. It MAY change when a [non-modifying operation](https://github.com/cnabio/cnab-spec/blob/master/101-bundle-json.md#custom-actions) is performed on an installation.
```

Comment:
Regarding the change from "MUST NOT" to "MAY":

We are using the current wording of the spec to enforce that 

1. No concurrent executions of `modifies: true` actions are run for the same installation (linearizability for modifying actions)
2. Any `modifies: true` action invocations are queued until all running `stateless: false` actions at the current revision are completed or canceled
3. Any number of `stateless: false, modifies: false` actions are allowed to run concurrently as long as there are no `modifies: true` actions in progress
4. Any claims for `stateless: false` or `modifies: true` actions referencing revisions older than the latest are invalidated and cannot be used to start new operations
5. Claims for `stateless: true, modifies: false` actions are always valid and may be executed concurrently with any other action(s)

Another thought is that in workflows that execute multiple `modifies: false` actions against the same installation, the revision may be used as a key to detect conflicts in a "transaction" and abort or retry.

We should be careful to avoid the implementation details of any particular runtime from leaking unnecessarily into the spec. Certainly points 1-2 can be handled with the proposed change, but I'm worried that in the future enforcing these additional invariants may be problematic for runtimes that have committed to always updating revisions.